src/Playwright/Program.cs

```diff
@@ -0,0 +1,123 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.Playwright.Helpers;
+
+namespace Microsoft.Playwright
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var p = new Program();
+            p.Run(args);
+        }
+
+        public void Run(string[] args)
+        {
+            string path = Directory.GetCurrentDirectory();
+
+            if (args != null && args.Length > 1 && args[0] == "-p")
+            {
+                path = Path.Combine(Directory.GetCurrentDirectory(), args[1]);
+                var newArgs = new string[args.Length - 2];
+                for (var i = 0; i < args.Length; ++i)
+                {
+                    newArgs[i] = args[i + 2];
+                }
+
+                args = newArgs;
+            }
+
+            string pwPath = null;
+            try
+            {
+                pwPath = Paths.GetExecutablePath();
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Please install Playwright:");
+                Console.WriteLine("   dotnet add package Microsoft.Playwright");
```

Comment:
How could this be the case? We are running code from the Microsoft.Playwright already.

Reply:
This can only be the case if we found a Microsoft.Playwright.dll with no .playwright. I should ask to build instead.